### PR TITLE
k8s: update configs and docs for secure mode

### PR DESCRIPTION
**Pull request by mberhault**
* use request-certs docker image v0.2 (pending in https://github.com/cockroachdb/k8s/pull/2)
* pass `namespace` to request-certs
* use only 1 pod for the example app
* improve docs on cert approval process (once per pod for nodes, one for
each deployment for example app)

### CODE REVIEW

**Comment by mberhault** [comment]
CC: @jseldess: this now works better. Still some testing to do, and probably some clarification on requirements.
However, rescheduled pods and scaled-up clients will not require new certs, they'll use the existing ones.

File: cloud/kubernetes/README.md [hunk 1]
```diff
@@ -54,6 +77,8 @@ which will lead to a lot of trouble.
 Secure mode currently works by requesting node/client certificates from the kubernetes
 controller at pod initialization time.
 
+
+
```
Comment (a-robinson) [inline cloud/kubernetes/README.md]: Any reason for these newlines?

File: cloud/kubernetes/README.md [hunk 2]
```diff
@@ -186,11 +221,16 @@ If you created a secure cockroach cluster, run:
...
-For every pod being created, you will need to approve its client certificate request:
+When the first pod is being initialized, you will need to approve its client certificate request:
 ```shell
-kubectl certificate approve client.root-example-secure-etc..
+kubectl certificate approve client.root
 ```
 
+If more pods are then added through `kubectl scale deployment example-secure --replicas=X`, the generated
+certificate will be reused.
+**WARNING**: the example app in secure mode should be started with only one replica, or concurrent and
```
Comment (a-robinson) [inline cloud/kubernetes/README.md]: Any plans to fix this?

**Review by a-robinson — APPROVED** [review]
LGTM

File: cloud/kubernetes/README.md (hunk 1 above)
Comment (mberhault) [inline cloud/kubernetes/README.md]: uh. nope. Removed.

File: cloud/kubernetes/README.md (hunk 2 above)
Comment (mberhault) [inline cloud/kubernetes/README.md]: I was thinking of adding a simple example which would just be the cockroach binary with `root` client certs and use that to connect to the cluster. This would replace the insecure invocation of `kubectl exec ...` as well. How does that sound?

Comment (a-robinson) [inline cloud/kubernetes/README.md]: That sounds fine, but seems orthogonal to the issue of concurrent cert requests by `cockroachdb/cockroach-k8s-request-cert` causing issues

Comment (mberhault) [inline cloud/kubernetes/README.md]: Oh, sorry, I stopped reading after seeing the `**WARNING**` line and thought you were pointing to the non-working cockroach client invocation in secure mode. Bleh!
I could fix this by either
* using slightly different names in the CSR (but that's annoying if you start a bunch of client replicas, you would need to do something with each CSR)
* having a marker saying the certificate request is "in process" so other replicas could just watch for the secret (I would need something with atomic guarantees in k8s. eg: "create secret if not exists")
* some other type of sharing?

Comment (a-robinson) [inline cloud/kubernetes/README.md]: How does the failure manifest? I'd think that we could just detect an "already exists" error when trying to create the CSR, but does the Kubernetes API not return such an error in that case?

Comment (mberhault) [inline cloud/kubernetes/README.md]: It does and it's trivial to detect. I could use that as a sign that I should be watching the secret instead.

Comment (a-robinson) [inline cloud/kubernetes/README.md]: Yeah, I'd probably go with something like that. Although to really be safe I think you'll need a bit more logic. It's possible that the CSR could have been rejected, or that it was approved but the client that created the CSR was taken down before it could create the secret with it so the secret will never be created. You could probably do something like:

1. Try to create CSR. If it creates successfully, wait for CSR to be approved.
1. If it already exists, get it. If it's still pending, wait for it to be approved.
1. If the previous CSR was rejected, create a new CSR (by deleting the old one I guess?) and wait for it to be approved.
1. If the previous CSR was already approved, go ahead and create the secret for it.

Comment (mberhault) [inline cloud/kubernetes/README.md]: I can definitely do 1. and 2. I think 3. should just cause any client to bail out, or we may be spamming with new CSRs when they've been intentionally denied. 4. can't work without storing the key of the original requester.
